tests/runners/bulk_update_runner.py

```diff
@@ -0,0 +1,138 @@
+import os
+import time
+from copy import deepcopy
+
+from openpyxl import load_workbook, Workbook
+from openpyxl.worksheet.worksheet import Worksheet
+
+from tests.ingest_agents import IngestBrokerAgent, IngestApiAgent
+from tests.runners.bulk_update_manager import BulkUpdateManager
+from tests.runners.submission_manager import SubmissionManager
+from tests.utils import Progress
+
+MODIFIED_BSD_ACCESSION_ID = '999'
+
+UI_CHANGE_VALUE = 'UI CHANGE '
+
+
+class BulkUpdateRunner:
+    def __init__(self, ingest_broker: IngestBrokerAgent, ingest_api: IngestApiAgent,
+                 bulk_update_manager: BulkUpdateManager):
+        self.ingest_broker = ingest_broker
+        self.ingest_api = ingest_api
+        self.submission_id = None
+        self.project_id = None
+        self.biomaterial_ids = None
+        self.submission_envelope = None
+        self.dataset = None
+        self.submission_manager: SubmissionManager = None
+        self.bulk_update_manager = bulk_update_manager
+
+    def bulk_update_run(self, dataset_fixture):
+        self.__import_spreadsheet_with_files(dataset_fixture)
+        self.__get_original_content()
+        updated_project_description, updated_contributor_name, updated_bsd_accession = \
+            self.__modify_metadata_with_api_calls()
+        update_spreadsheet, updated_spreadsheet_path = self.__download_modified_spreadsheet()
+        updated_project_title, updated_biomaterial_name, modified_biomaterial_id = \
+            self.__modify_metadata_in_sheet(update_spreadsheet)
+        self.bulk_update_manager.save_modified_spreadsheet(update_spreadsheet, updated_spreadsheet_path)
+        self.__upload_modified_spreadsheet(updated_spreadsheet_path)
+        self.__validate_modifications(updated_project_description, updated_contributor_name, updated_bsd_accession,
+                                      updated_project_title, updated_biomaterial_name)
+
+    def __import_spreadsheet_with_files(self, dataset_fixture, project_uuid=None):
+        self.dataset = dataset_fixture
+        self.__upload_spreadsheet_and_create_submission(dataset_fixture, project_uuid)
+        self.submission_manager = SubmissionManager(self.submission_envelope)
+        self.submission_manager.get_upload_area_credentials()
+        self.submission_manager.stage_data_files(self.dataset.config['data_files_upload_area_uuid'])
```

Comment:
I believe you do not need to upload files to test metadata updates

Reply:
removing it will also save time for running this test

Reply:
Are you sure? It only take 1-2 seconds

Reply:
yeah the test files are very small, i guess no big impact/not much processing is saved if you remove this